### DISCUSSION TITLE
Entity extension visibility in admin and storefront

### DISCUSSION THREAD

**Discussion by bogdannistorbogdan** [medusajs/medusa#5686] [Q&A]
Hello! I want to extend the product entity by adding a new field and I followed the instructions from here:
https://docs.medusajs.com/development/entities/extend-entity
and here:
https://docs.medusajs.com/development/entities/migrations/create

I can see the newly added property in the database. I extended the "Product" entity and added a new field, "Brand". Now, how can I have this "Brand" show up in the Admin interface and, obviously, in the store frontend? I am obviously missing the link between what I did and the flow of having the backend and frontend actually show this field.
Also what if I want to completely ignore some fields, like for example, "material". It's in the product table. I really don't need this field for my use case.
Thanks.

**Comment by alightinastorm** [comment]
You can add additional widgets to the admin ui, by using widgets
https://docs.medusajs.com/admin/widgets
The injection zone you are likely looking for is: "product.details.after" or "product.details.before"

To make the ui consistent, use medusa-ui
https://docs.medusajs.com/ui

You might have to create a custom API-Route to save the data.
To use it in a storefront, change the code in your storefront

Afaik, there is no way to remove fields from the base

**Reply by bogdannistorbogdan** [reply]
Thank you! I'll try this approach.

**Reply by JasonBrardDev** [reply]
@robodove Do you know how we can add a custom field input on the product create/edit page? 
Currently, the injection zone only correspond to the product detail page.

**Reply by alightinastorm** [reply]
> @robodove Do you know how we can add a custom field input on the product create/edit page? 
> 
> Currently, the injection zone only correspond to the product detail page.

Not possible currently, unless you eject and copy the whole admin frontend and cistomize it yourself 

There's a discussion here about that (admin customization) and the medusa team has put that on the roadmap, though nothing shipped so far

**Reply by JasonBrardDev** [reply]
Arf, too bad !
Thank you for your answer.

**Comment by dibs439** [comment]
How can I being the new custom field in edit product popup?
